components/org.wso2.carbon.identity.application.authz.xacml/src/test/java/org.wso2.carbon.identity.application.authz.xacml/handler.impl/XACMLBasedAuthorizationHandlerTest.java

```diff
@@ -0,0 +1,169 @@
+import org.wso2.carbon.identity.application.authentication.framework.model.AuthenticatedUser;
+import org.wso2.carbon.identity.application.authentication.framework.util.FrameworkUtils;
+import org.wso2.carbon.identity.application.authz.xacml.internal.AppAuthzDataholder;
+import org.wso2.carbon.identity.entitlement.EntitlementException;
+import org.wso2.carbon.identity.entitlement.EntitlementService;
+import org.wso2.carbon.identity.entitlement.common.dto.RequestDTO;
+import org.wso2.carbon.identity.entitlement.common.util.PolicyCreatorUtil;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyString;
+import static org.powermock.api.mockito.PowerMockito.doNothing;
+import static org.powermock.api.mockito.PowerMockito.mock;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+import static org.powermock.api.mockito.PowerMockito.spy;
+import static org.powermock.api.mockito.PowerMockito.when;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertNotNull;
+import static org.testng.Assert.assertTrue;
+import static org.testng.AssertJUnit.assertEquals;
+
+/**
+ * XACMLBasedAuthorizationHandlerTest defines unit tests for XACMLBasedAuthorizationHandler class.
+ */
+@PrepareForTest({LogFactory.class, FrameworkUtils.class, PolicyCreatorUtil.class, PolicyBuilder.class})
+@PowerMockIgnore("javax.xml.*")
+public class XACMLBasedAuthorizationHandlerTest {
+
+    private XACMLBasedAuthorizationHandler xacmlBasedAuthorizationHandler;
+    private AuthenticationContext context;
+    private Log log = mock(Log.class);
+
+    private static final String ADMIN_USER = "admin_user";
+    private static final String DECISION = "decision";
+    private static final String RULE_EFFECT_PERMIT = "Permit";
+    private static final String RULE_EFFECT_NOT_APPLICABLE = "NotApplicable";
+    private static final String POLICY = "policy";
+    private static final String ERROR = "error";
+    private static String XACML_RESPONCE = "<ns:root xmlns:ns=\"urn:oasis:names:tc:xacml:3.0:core:schema:wd-17\">"
+            +   "<ns:Result>"
+            +   "<ns:Decision>"
+            +   DECISION
+            +   "</ns:Decision>"
+            +   "</ns:Result>"
+            +   "</ns:root>";
+
+    @ObjectFactory
+    public IObjectFactory getObjectFactory() {
+
+        return new PowerMockObjectFactory();
+    }
+
+    @BeforeClass
+    public void init() {
+
+        mockStatic(LogFactory.class);
+        when(LogFactory.getLog(XACMLBasedAuthorizationHandler.class)).thenReturn(log);
+        xacmlBasedAuthorizationHandler = spy(XACMLBasedAuthorizationHandler.getInstance());
+        context = mock(AuthenticationContext.class);
+        when(log.isDebugEnabled()).thenReturn(true);
+    }
+
+    @Test
+    public void testGetInstance() {
```

Comment:
getAnother instance and check whether the two instance are equal

Reply:
done